scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java

```diff
@@ -47,7 +47,12 @@
      */
     public Protos.TaskInfo createTask(Configuration configuration, FrameworkState frameworkState, Protos.Offer offer) {
         this.frameworkState = frameworkState;
-        List<Integer> ports = Resources.selectTwoPortsFromRange(offer.getResourcesList());
+        List<Integer> ports;
+        if (configuration.getElasticsearchPorts().isEmpty()) {
+            ports = Resources.selectTwoPortsFromRange(offer.getResourcesList());
+        } else {
+            ports = configuration.getElasticsearchPorts();
+        }
```

Comment:
I'm worried about this logic. User want to limits ports, however they still should be in the offer, aren't they? Should we filter ports in the offer through `configuration.getElasticsearchPorts()` and then select two?